core/arch/arm/plat-imx/sm_platform_handler.c

```diff
@@ -41,6 +43,38 @@ static bool imx_sip_handler(struct thread_smc_args *smc_args)
 	return false;
 }
 
+#ifdef CFG_SM_PLATFORM_MBED_LINUX
+static void imx_oem_populate_fdt(struct thread_smc_args *smc_args)
+{
+	/* Populate with generic_boot provided parameters */
+	init_fdt(smc_args->a1);
```

Comment:
This address need to be checked. What if it's a secure address?

Reply:
I thought of that and tested malicious input.
It behaves the same way as giving an invalid DTB pointer as an OPTEE arg.

=> smc 0x83000001 0
E/TC:0   init_fdt:786 Device Tree missing
Res:  0 0 0 0